pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -1342,4 +1342,24 @@ self: super: builtins.intersectAttrs super {
       gi-webkit2
       gi-webkit2webextension
       ;
+
+  # Makes the mpi-hs package respect the choice of mpi implementation in Nixpkgs.
+  # Also adds required test dependencies for checks to pass
+  mpi-hs =
+    let validMpi = [ "openmpi" "mpich" "mvapich" ];
+        mpiImpl = pkgs.mpi.pname;
+        disableUnused = with builtins; map disableCabalFlag (filter (n: n != mpiImpl) validMpi);
+     in lib.pipe
+          (super.mpi-hs_0_7_3_0.override { ompi = pkgs.mpi; })
+          ( [ (addTestToolDepends [ pkgs.openssh pkgs.mpiCheckPhaseHook ]) ]
+            ++ disableUnused
+            ++ lib.optional (builtins.elem mpiImpl validMpi) (enableCabalFlag mpiImpl)
+          );
```

Comment:
I am wondering whether this code is a bit to general in the mpiImpl. If the lib.optional case fails the package will be basically unusable, won’t it? Doesn’t that mean it would better to fail loudly in that case?

Reply:
Luckily not! :) If none of the flags is true, mpi-hs will try to generically link against libmpi, see here https://github.com/eschnett/mpi-hs/blob/master/package.yaml#L89
This actually works for all three MPI implementations, too. If even this **would** fail (which it luckily doesn't) the test suite would not pass.